superset/assets/javascripts/dashboard/components/DashboardViewContainer.jsx

```diff
@@ -0,0 +1,305 @@
+      return f;
+    }
+
+    // Building a list of fields the slice is immune to filters on
+    let immuneToFields = [];
+    if (
+      sliceId &&
+      metadata.filter_immune_slice_fields &&
+      metadata.filter_immune_slice_fields[sliceId]) {
+      immuneToFields = metadata.filter_immune_slice_fields[sliceId];
+    }
+    for (const filteringSliceId in filters) {
+      if (filteringSliceId === sliceId.toString()) {
+        // Filters applied by the slice don't apply to itself
+        continue;
+      }
+      for (const field in filters[filteringSliceId]) {
+        if (!immuneToFields.includes(field)) {
+          f.push({
+            col: field,
+            op: 'in',
+            val: filters[filteringSliceId][field],
+          });
+        }
+      }
+    }
+    return f;
+  }
+
+  jsonEndpoint(data, force = false) {
+    let endpoint = getExploreUrl(data, 'json', force);
+    if (endpoint.charAt(0) !== '/') {
+      // Known issue for IE <= 11:
+      // https://connect.microsoft.com/IE/feedbackdetail/view/1002846/pathname-incorrect-for-out-of-document-elements
+      endpoint = '/' + endpoint;
+    }
+    return endpoint;
+  }
+
+  loadPreSelectFilters() {
+    for (const key in this.props.filters) {
+      for (const col in this.props.filters[key]) {
+        const sliceId = parseInt(key, 10);
+        this.props.actions.addFilter(sliceId, col,
+          this.props.filters[key][col], false, false);
+      }
+    }
+  }
+
+  refreshExcept(sliceId) {
+    const immune = this.props.dashboard.metadata.filter_immune_slices || [];
+    const slices = this.getAllSlices()
+      .filter(slice => slice.slice_id !== sliceId && immune.indexOf(slice.slice_id) === -1);
+    this.renderSlices(slices);
+  }
+
+  updateFilterParamsInUrl() {
+    const urlObj = urlLib.parse(location.href, true);
+    urlObj.query = urlObj.query || {};
+    urlObj.query.preselect_filters = this.readFilters();
+    urlObj.search = null;
+    history.pushState(urlObj.query, window.title, urlLib.format(urlObj));
+  }
+
+  stopPeriodicRender() {
+    if (this.refreshTimer) {
+      clearTimeout(this.refreshTimer);
+      this.refreshTimer = null;
+    }
+  }
+
+  startPeriodicRender(interval) {
+    this.stopPeriodicRender();
+    const dash = this;
+    const immune = this.props.dashboard.metadata.timed_refresh_immune_slices || [];
+    const refreshAll = () => {
+      const affectedSlices = this.getAllSlices()
+        .filter(slice => immune.indexOf(slice.slice_id) === -1);
+      dash.renderSlices(affectedSlices, true, interval * 0.2);
+    };
+    const fetchAndRender = function () {
+      refreshAll();
+      if (interval > 0) {
+        dash.refreshTimer = setTimeout(fetchAndRender, interval);
+      }
+    };
+
+    fetchAndRender();
+  }
+
+  readFilters() {
+    // Returns a list of human readable active filters
+    return JSON.stringify(this.props.filters, null, '  ');
+  }
+
+  updateDashboardTitle(title) {
```

Comment:
I'm trying to understand why these methods that match actions 1:1 are needed? Seems like this is boilerplate/unneeded redirection...